hostmap.go

```diff
@@ -242,6 +242,10 @@ type HostInfo struct {
 	// Used to track other hostinfos for this vpn ip since only 1 can be primary
 	// Synchronised via hostmap lock and not the hostinfo lock.
 	next, prev *HostInfo
+
+	//TODO: in, out, and others might benefit from being an atomic.Int32. We could collapse connectionManager pendingDeletion, relayUsed, and in/out into this 1 thing
+	in, out, pendingDeletion atomic.Bool
+	lastComms                atomic.Pointer[time.Time]
```

Comment:
Does this result in a lot of allocations? I think the time.Time pointer must be allocated and is modified per packet in/out?

I wonder if we can use something cheaper here, like an atomic int. I bet we could use something from the timer wheel to track the advancement of time without being so exact as a time.Time struct.